fulp_modules/main_features/bloodsuckers/code/powers/feed.dm

```diff
@@ -194,133 +210,119 @@
-			break
+	///////////////////////////////////////////////////////////
+	// MOOD EFFECTS //
+	// Drank good blood? - GOOD
+	if(amount_taken > 5 && target.stat < DEAD && ishuman(target))
+		SEND_SIGNAL(user, COMSIG_ADD_MOOD_EVENT, "drankblood", /datum/mood_event/drankblood)
+	// Drank mindless as Ventrue? - BAD
+	if(!target.mind && bloodsuckerdatum?.my_clan == CLAN_VENTRUE)
+		SEND_SIGNAL(user, COMSIG_ADD_MOOD_EVENT, "drankblood", /datum/mood_event/drankblood_bad)
+		if(!warning_target_inhuman)
+			to_chat(user, "<span class='notice'>You feel disgusted at the taste of a non-sentient creature.</span>")
+			warning_target_inhuman = TRUE
+	// Dead Blood? - BAD
+	if(target.stat >= DEAD)
+		if(ishuman(target))
+			SEND_SIGNAL(user, COMSIG_ADD_MOOD_EVENT, "drankblood", /datum/mood_event/drankblood_dead)
+		if(!warning_target_dead)
+			to_chat(user, "<span class='notice'>Your victim is dead. [target.p_their(TRUE)] blood barely nourishes you.</span>")
+			warning_target_dead = TRUE
+	// Drank off of a non-carbon? - BAD
+	if(!ishuman(target))
+		SEND_SIGNAL(user, COMSIG_ADD_MOOD_EVENT, "drankblood", /datum/mood_event/drankblood_bad)
+		if(!warning_target_inhuman)
+			to_chat(user, "<span class='notice'>You recoil at the taste of a lesser lifeform.</span>")
+			warning_target_inhuman = TRUE
 
-		// Blood Gulp Sound
-		owner.playsound_local(null, 'sound/effects/singlebeat.ogg', 40, TRUE)
-		if(!amSilent)
-			target.playsound_local(null, 'sound/effects/singlebeat.ogg', 40, TRUE)
-	// DONE!
-	//DeactivatePower(user,target)
-	if(amSilent)
-		to_chat(user, "<span class='notice'>You slowly release [target]'s wrist." + (target.stat == 0 ? " [target.p_their(TRUE)] face lacks expression, like you've already been forgotten.</span>" : ""))
-	else
-		user.visible_message("<span class='warning'>[user] unclenches their teeth from [target]'s neck.</span>", \
-							 "<span class='warning'>You retract your fangs and release [target] from your bite.</span>")
-		REMOVE_TRAIT(user, TRAIT_IMMOBILIZED, BLOODSUCKER_TRAIT)
-	bloodsuckerdatum.total_blood_drank += amount_taken
-	log_combat(owner, target, "fed on blood", addition="(and took [amount_taken] blood)")
+	// Blood Remaining? (Carbons/Humans only)
+	else if(!IS_BLOODSUCKER(target))
+		if(target.blood_volume <= BLOOD_VOLUME_BAD && warning_target_bloodvol > BLOOD_VOLUME_BAD)
+			to_chat(user, "<span class='warning'>Your victim's blood volume is fatally low!</span>")
+		else if(target.blood_volume <= BLOOD_VOLUME_OKAY && warning_target_bloodvol > BLOOD_VOLUME_OKAY)
+			to_chat(user, "<span class='warning'>Your victim's blood volume is dangerously low.</span>")
+		else if(target.blood_volume <= BLOOD_VOLUME_SAFE && warning_target_bloodvol > BLOOD_VOLUME_SAFE)
+			to_chat(user, "<span class='notice'>Your victim's blood is at an unsafe level.</span>")
+		warning_target_bloodvol = target.blood_volume // If we had a warning to give, it's been given by now.
```

Comment:
this never gets reset back to 99999.

this is a pretty hacky way to do this, btw- is there any other way you can think of to do this?